lib/mixins/property-effects.html

```diff
@@ -411,32 +416,122 @@
+    if (property !== TRANSITIVE_DEPENDENCY || 
+        computedPropertyIsDirty(inst, info, changedProps)) {
+      let result = runMethodEffect(inst, property, changedProps, oldProps, info);
+      let computedProp = info.methodInfo;
+      if (inst.__dataHasAccessor && inst.__dataHasAccessor[computedProp]) {
+        inst._setPendingProperty(computedProp, result, true);
+      } else {
+        inst[computedProp] = result;
+      }
     }
   }
+  
+  /**
+   * Returns any dependencies of a computed property that are themselves
+   * also computed.
+   *
+   * @param {!PropertyEffectsType} inst The instance to test
+   * @param {?} info Computed effect metadata
+   * @return {Array<?>} Array of computed effect metadata for depenencies
+   * @private
+   */
+   function computedDependenciesFor(inst, info) {
+    let deps = info.computedDependencies;
+    if (!deps) {
+      deps = info.computedDependencies = [];
+      info.args.forEach(arg => {
+        const dep = arg.rootProperty || arg.name;
+        const depInfo = inst[COMPUTE_INFO][dep];
+        if (depInfo) {
+          deps.push(depInfo);
+        }
+      });
+    }
+    return deps;
+  }
+
+  /**
+   * Runs computed effects for dependencies of the computed property effect
+   * described by `info`.
+   *
+   * @param {!PropertyEffectsType} inst The instance to test
+   * @param {Object} info Effect metadata
+   * @param {Object} changedProps Bag of current property changes
+   * @param {Object} oldProps Bag of previous values for changed properties
+   * @return {void}
+   * @private
+   */
+   function computeDependencies(inst, info, changedProps, oldProps) {
+    let deps = computedDependenciesFor(inst, info);
+    if (deps.length) {
+      deps.forEach(depInfo => {
+        if (depInfo.lastRun !== info.lastRun) {
+          depInfo.lastRun = info.lastRun;
+          runComputedEffect(inst, TRANSITIVE_DEPENDENCY, changedProps, oldProps, depInfo);
+        }
+      });
+    }
+  }
+
+  /**
+   * Returns whether the computed effect has any changed dependencies.
+   *
+   * @param {!PropertyEffectsType} inst The instance to test
+   * @param {?} info Effect metadata
+   * @param {Object} changedProps Bag of current property changes
+   * @return {boolean} true when the computed effect should be re-calculated
+   * @private
+   */
+  function computedPropertyIsDirty(inst, info, changedProps) {
+    return info.dynamicFn && propertyIsDirty(inst, {name: info.methodName}, changedProps) ||
+      info.args.some(arg => propertyIsDirty(inst, arg, changedProps));
+  }
+
+  /**
+   * Returns whether any changed props match the effect trigger
+   *
+   * @param {!PropertyEffectsType} inst The instance to test
+   * @param {DataTrigger} trigger Descriptor
+   * @param {Object} changedProps Bag of current property changes
+   * @return {boolean} true when the property is dirty
+   * @private
+   */
+  function propertyIsDirty(inst, trigger, changedProps) {
```

Comment:
I would put these functions before `runComputedEffect`. That makes reading these functions a bit easier, since `runComputedEffect` calls into these functions and I was confused which ones they were.